### DISCUSSION TITLE
Amasty Product Parts Finder

### DISCUSSION THREAD

**Discussion by techweb-digimedia** [breezefront/community#162] [3. Integrations]
Hello, I encountered a compatibility issue with the Amasty module "Products part finder" (https://amasty.com/product-parts-finder-for-magento-2.html): js events are not fired on dropdown change.
I have already contacted Amasty support if they could fix the problem and they replied saying that the theme disables all third party scripts by default including the scripts needed for the module to work, they tried following this guide to enqueue module scripts but without success (https://docs.swissuplabs.com/m2/extensions/breeze/devdocs/module-js/).
Could you help us make the integration work? 
Thanks

**Comment by vovayatsyuk** [comment]
Do you have a non-working integration from Amasty? Do you have a store available so I could see what have you done already and why it's not working?

**Reply by techweb-digimedia** [reply]
Here you can find an example: https://kitefront.digimedia.it/ruote-e-accessori/ruote.html
No errors are shown in console

Here where the missing files should be placed as indicated by the module developers:
![2023-06-17_19 58 11](https://github.com/breezefront/community/assets/137304546/865b2c3c-0ed4-4004-a07c-97219389c0d8)

These files that the module needs to work:
amfinder.js
amFinderAjaxOptions.js
chosen.jquery.min.js
notification.js

After the module developer's attempt, I also tried to replicate the process in the guide above by creating "breeze_default.xml" with the following content

```
<?xml version="1.0"?>
<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
        <referenceBlock name="breeze.js">
            <arguments>
                <argument name="bundles" xsi:type="array">
                    <item name="default" xsi:type="array">
                        <item name="items" xsi:type="array">
                            <item name="Amasty_Finder/js/amfinder" xsi:type="string">Amasty_Finder/js/breeze/amfinder</item>
                            <item name="Amasty_Finder/js/amFinderAjaxOptions" xsi:type="string">Amasty_Finder/js/breeze/amFinderAjaxOptions</item>
                            <item name="Amasty_Finder/js/chosen.jquery.min" xsi:type="string">Amasty_Finder/js/breeze/chosen.jquery.min</item>
                            <item name="Amasty_Finder/js/notification" xsi:type="string">Amasty_Finder/js/breeze/notification</item>
                        </item>
                    </item>
                </argument>
            </arguments>
        </referenceBlock>
    </body>
</page>
```

Is it correct? Should it be placed in the module folder as mentioned in the docs (https://breezefront.com/docs/custom-module) or could it be placed in a child theme (Local/breeze-custom)?

Thanks

**Reply by vovayatsyuk** [reply]
Disable js merge during development, so each js file will be included separately.

This module uses an inline script to initialize itself: 

```html
<script type="text/javascript">
  require(['jquery', 'Amasty_Finder/js/amfinder'], function ($) {
    var finderConfig = {"ajaxUrl":"..."};
    
    $("#" + finderConfig.containerId).amfinder(finderConfig);
  });
</script>
```

Breeze removes these types of scripts by default. You need to add a `data-breeze` attribute to the script tag so we keep it in the document:

`<script data-breeze type="text/javascript">`

**Reply by techweb-digimedia** [reply]
Disabled js merge and added data-breeze to script tag, now the console shows "Uncaught TypeError: $(...).amfinder is not a function" and i can't find the module scripts in the source tab, is breeze_default.xml correct? Where it should be placed?

**Reply by vovayatsyuk** [reply]
This is not the only error in the console. The other errors tell that there are no such files:

```
<item name="Amasty_Finder/js/amfinder" xsi:type="string">Amasty_Finder/js/breeze/amfinder</item>
<item name="Amasty_Finder/js/amFinderAjaxOptions" xsi:type="string">Amasty_Finder/js/breeze/amFinderAjaxOptions</item>
<item name="Amasty_Finder/js/chosen.jquery.min" xsi:type="string">Amasty_Finder/js/breeze/chosen.jquery.min</item>
<item name="Amasty_Finder/js/notification" xsi:type="string">Amasty_Finder/js/breeze/notification</item>
```

All of these files should exist.

Take a look at other integrations for more examples: 

- https://github.com/breezefront/module-breeze-magefan-blog
- https://github.com/breezefront/module-breeze-mageworx-advancedproductoptions
- https://github.com/breezefront/module-breeze-amasty-xsearch

**Reply by techweb-digimedia** [reply]
File paths were actually wrong (Amasty_Finder/js/~~breeze~~/amfinder), now the scripts are enqueued correctly.
Unfortunately i also encountered some incompatibilities with cash.js, module scripts and jquery Chosen plugin (which can be disabled in the module setting) but i managed to find some workarounds and now it works, thanks for your help!

**Reply by johndiesel123** [reply]
> Unfortunately i also encountered some incompatibilities with cash.js, module scripts and jquery Chosen plugin (which can be disabled in the module setting) but i managed to find some workarounds and now it works, thanks for your help!

 Can you share the steps you took to resolve this? I'm attempting to resolve the same issue where the parts finder jQuery/ajax not working when you select the first parts finder dropdown item.
